rust/operator-binary/src/csi_server/node.rs

```diff
@@ -223,10 +244,26 @@ impl SecretProvisionerNode {
         };
         for (k, v) in data
             .data
-            .into_files(format, compat)
+            .into_files(format, names, compat)
             .context(publish_error::FormatDataSnafu)?
         {
             let item_path = target_path.join(k);
+
+            // Prevent unwanted path traversals by first canonicalizing the final
+            // path and then validating that the path starts with the base we
+            // expect.
+            let item_path = item_path
+                .canonicalize()
+                .context(publish_error::CanonicalizePathSnafu { path: &item_path })?;
```

Comment:
This looks like more fodder for "we really should just be doing #345 instead (with NOFOLLOW)".

Reply:
I think I'm okay with letting this in for 25.3.0 since the same problem already exists for k8sSearch, so it's not technically a regression...

Reply:
Alright, your call. I'm happy to move forward as is and look into `openat` after the release is done.